src/expr/src/relation/mod.rs

```diff
@@ -2769,7 +2769,23 @@ impl RowSetFinishing {
             })
             .unwrap_or(limit);
 
-        let mut ret = Vec::with_capacity(return_size);
+        // Bail early if we know our results won't fit into our Vec
+        // 
+        // Note: this check is a _lower bound_ as to how much memory
+        // we'll need for results. Calculating the actual answer would
+        // require us to iterate through all of the rows
+        if return_size * std::mem::size_of::<Row>() > max_result_size {
+            return Err(format!(
+                "result would exceed max size of {}",
+                ByteSize::b(u64::cast_from(max_result_size))
+            ));
+        }
+
+        // The return Vec can sometimes be quite large, so we'll use the
+        // failable allocation APIs to prevent a panic
+        let mut ret = Vec::new();
+        ret.try_reserve(return_size).map_err(|e| e.to_string())?;
```

Comment:
I don't agree with this reasoning. In general, any time environmentd OOMs, it is considered a bug. I would rather we panic loudly here to detect OOMs if indeed there are any.

Also, failed allocations crashes, OOMs, and panics are three different things, so even if you kept this logic, you would need to change the comment.

```
~/test_rust ❯❯❯ cat src/bin/test_rust.rs                                                                       ✘ 134 
fn main() {
    let mut v = Vec::<usize>::with_capacity(1_000_000_000_000_000);

    v[100_000_000_000] = 42;

    println!("{v:?}");
}
~/test_rust ❯❯❯ cargo run --release
warning: unused manifest key: lints
    Finished release [optimized] target(s) in 0.03s
     Running `target/release/test_rust`
memory allocation of 8000000000000000 bytes failed
[1]    514248 IOT instruction  cargo run --release
```

Not sure what to call this, but it's not a panic in the Rust sense.

Reply:
If you don't want to panic, we should at the very least log with `tracing::error` (someday soon, I hope, we will make it so that errors ping the on-call)

Reply:
Makes sense to me! I returned this to the prior state where we'll panic if we fail to allocate